cloud/microsoft/office365/exchange/mode/emailactivity.pm

```diff
@@ -187,36 +191,43 @@ sub manage_selection {
     
     $self->{active} = { active => 0, total => 0, report_date => '' };
     $self->{global} = { send_count => 0, receive_count => 0 , read_count => 0 };
-    $self->{mailboxes} = {};
+    $self->{users} = {};
+
+    my $results = $options{custom}->office_get_exchange_activity(param => "period='D7'");
+    my $results_daily = [];
+    if (scalar(@{$results})) {
+       $self->{active}->{report_date} = @{$results}[0]->{'Report Refresh Date'};
+       $results_daily = $options{custom}->office_get_exchange_activity(param => "date=" . $self->{active}->{report_date});
+    }
 
-    my $results = $options{custom}->office_get_exchange_activity();
+    foreach my $user (@{$results}, @{$results_daily}) {
+        # Let's lc the instance label to make metrics "clean"...
+        $user->{'User Principal Name'} = lc($user->{'User Principal Name'});
 
-    foreach my $mailbox (@{$results}) {
-        if (defined($self->{option_results}->{filter_mailbox}) && $self->{option_results}->{filter_mailbox} ne '' &&
-            $mailbox->{'User Principal Name'} !~ /$self->{option_results}->{filter_mailbox}/) {
-            $self->{output}->output_add(long_msg => "skipping  '" . $mailbox->{'User Principal Name'} . "': no matching filter name.", debug => 1);
+        if (defined($self->{option_results}->{filter_user}) && $self->{option_results}->{filter_user} ne '' &&
+            $user->{'User Principal Name'} !~ /$self->{option_results}->{filter_user}/) {
+            $self->{output}->output_add(long_msg => "skipping  '" . $user->{'User Principal Name'} . "': no matching filter name.", debug => 1);
             next;
         }
-    
-        $self->{active}->{total}++;
 
-        if (!defined($mailbox->{'Last Activity Date'}) || $mailbox->{'Last Activity Date'} eq '' ||
-            ($mailbox->{'Last Activity Date'} ne $mailbox->{'Report Refresh Date'})) {
-            $self->{output}->output_add(long_msg => "skipping '" . $mailbox->{'User Principal Name'} . "': no activity.", debug => 1);
+        if ($user->{'Report Period'} != 1) {
+            if (!defined($user->{'Last Activity Date'}) || ($user->{'Last Activity Date'} ne $self->{active}->{report_date})) {
+                $self->{output}->output_add(long_msg => "skipping '" . $user->{'User Principal Name'} . "': no activity.", debug => 1);
+            }
+            $self->{active}->{total}++;
```

Comment:
This mean it excludes every accounts that never had an activity?
That doesn't suite me. I prefer adding another metric like active/inactive/total.

Reply:
No, I think you did not catch the algorithm principle. Let me explain.

For every mode, we do 2 requests :
- a `period` request (`D7`, so 7 days), which returns the number of active and inactive users (or sites etc...), aggregated metrics over the period, and the _Report Refresh Date_ ;
- a `date` request, at the date of _Report Refresh Date_ received from the previous request, which returns metrics for the active users (or sites...) at this specific date.

Thanks to the first call, we then can still compute the number of active and inactive users, as before.
And thanks to the second call, we compute much more precise per-user (or per-site...) metrics, because not aggregated over 7 days, but over 1 precise day only. Which is much more relevant, especially on graphs...

So in this piece of code (longer quote) :
https://github.com/centreon/centreon-plugins/blob/37172649f9380818b0f711516ad35be94319528b/cloud/microsoft/office365/exchange/mode/emailactivity.pm#L213-L219
We compute the total number of accounts, thanks to the `period` results (`Report Period == 7`).
And, thanks to the `next` instruction, we skip the code below, which concerns active users (or sites...) only, having their accurate data in `date` results (`Report Period == 1`).

So nothing wrong / missing here :)

About the 2 calls scheme. We can't directly do the second (`date`) call, as we don't know the last refresh date (which varies and is sometimes a few days ago). And in addition, this second call does not return inactive objects information. We then need both of them. But does not impact performances as explained [above](https://github.com/centreon/centreon-plugins/pull/2042#discussion_r462133126).
This answers your first point [here](https://github.com/centreon/centreon-plugins/pull/2042#pullrequestreview-457548963).